courses-rest/src/test/java/no/sikt/nva/fs/CoursesByInstitutionOfLoggedInUserHandlerTest.java

```diff
@@ -1,65 +1,284 @@
     private Context context;
     private CoursesByInstitutionOfLoggedInUserHandler handler;
     private ByteArrayOutputStream output;
 
     @BeforeEach
-    public void init() {
-        var environment = mock(Environment.class);
+    public void init(final WireMockRuntimeInfo wmRuntimeInfo) {
         when(environment.readEnv(ApiGatewayHandler.ALLOWED_ORIGIN_ENV)).thenReturn("*");
 
-        context = mock(Context.class);
+        final String fsBaseUri = UriWrapper.fromUri(wmRuntimeInfo.getHttpBaseUrl()).toString();
+        final String fsConfigString = IoUtils.stringFromResources(Path.of("fsConfig.json"))
+                                          .replace("@@BASE_URI@@", fsBaseUri);
+        when(environment.readEnvOpt(CoursesByInstitutionOfLoggedInUserHandler.FS_CONFIG_ENV_NAME))
+            .thenReturn(Optional.of(fsConfigString));
 
+        context = mock(Context.class);
         output = new ByteArrayOutputStream();
+    }
+
+    @Test
+    void shouldReturnEmptyListOfCoursesAndLogProblemIfFsIsUnavailable() throws IOException {
+        // prepare:
+        final InputStream input = createRequest(NON_SUPPORTED_INSTITUTION_PATH);
+        this.handler = new CoursesByInstitutionOfLoggedInUserHandler(environment, AFTER_SUMMER);
+
+        // execute:
+        handler.handleRequest(input, output, context);
+
+        // verify:
+        var gatewayResponse = GatewayResponse.fromOutputStream(output, Course[].class);
+
+        assertEquals(HttpURLConnection.HTTP_OK, gatewayResponse.getStatusCode());
+        assertThat(gatewayResponse.getHeaders(), hasKey(CONTENT_TYPE));
+        assertThat(gatewayResponse.getHeaders(), hasKey(ACCESS_CONTROL_ALLOW_ORIGIN));
+        assertThat(gatewayResponse.getBodyObject(Course[].class), emptyArray());
+    }
+
+    @Test
+    void shouldReturnEmptyListOfCoursesAndLogProblemIfFsReturnsInvalidData() throws IOException {
+        // prepare:
+        final InputStream input = createRequest(SUPPORTED_INSTITUTION_PATH);
+
+        final String responseBody = "[]";
+        stubRequestForCourses(2022, responseBody);
+
+        this.handler = new CoursesByInstitutionOfLoggedInUserHandler(environment, AFTER_SUMMER);
+
+        // execute:
+        handler.handleRequest(input, output, context);
 
-        this.handler = new CoursesByInstitutionOfLoggedInUserHandler(environment);
+        // verify:
+        var gatewayResponse = GatewayResponse.fromOutputStream(output, Course[].class);
+
+        assertEquals(HttpURLConnection.HTTP_OK, gatewayResponse.getStatusCode());
+        assertThat(gatewayResponse.getHeaders(), hasKey(CONTENT_TYPE));
+        assertThat(gatewayResponse.getHeaders(), hasKey(ACCESS_CONTROL_ALLOW_ORIGIN));
+        assertThat(gatewayResponse.getBodyObject(Course[].class), emptyArray());
     }
 
-    @AfterEach
-    public void tearDown() {
+    @Test
+    void shouldReturnEmptyListOfCoursesAndLogProblemIfFsReturnsNotAuthorized() throws IOException {
```

Comment:
Here the code should be 503 Service Unavailable or 511 Network Authentication Required. 

Returning OK in all problematic occasions makes problem resolution quite difficult. We should either fix it or create issues for fixing the responses in problematic cases

Reply:
It is due to how it works in DLR today. But I agree with your points made here. The problem should be propagated to the client and handled there.